scripts/launch/demo-cluster/default.nix

```diff
@@ -86,70 +102,82 @@ in pkgs.writeScript "demo-cluster" ''
+  '' else ''
+    echo "Creating genesis keys..."
+    config_files=${configFiles}
+    cardano-keygen --system-start 0 generate-keys-by-spec --genesis-out-dir ${stateDir}/genesis-keys --configuration-file $config_files/configuration.yaml --configuration-key ${configurationKey}
+  ''}
 
   trap "stop_cardano" INT TERM
   echo "Launching a demo cluster..."
   for i in {1..${builtins.toString numCoreNodes}}
   do
-    node_args="--db-path ${stateDir}/core-db''${i} --rebuild-db --genesis-secret ''${i} --listen 127.0.0.1:300''${i} --json-log ${stateDir}/logs/node''${i}.json --logs-prefix ${stateDir}/logs --system-start $system_start --metrics +RTS -N2 -qg -A1m -I0 -T -RTS --node-id core''${i} --topology ${topologyFile} --configuration-file ${configFiles}/configuration.yaml ${ifAssetLock "--asset-lock-file ${assetLockFile}"}"
-    echo Launching core node $i with args: $node_args
-    cardano-node-simple $node_args &> /dev/null &
+    node_args="--db-path ${stateDir}/core-db$i --rebuild-db ${if launchGenesis then "--keyfile ${stateDir}/genesis-keys/generated-keys/rich/key$((i - 1)).sk" else "--genesis-secret $i"} --listen 127.0.0.1:$((3000 + i)) --json-log ${stateDir}/logs/core$i.json --logs-prefix ${stateDir}/logs --system-start $system_start --metrics +RTS -N2 -qg -A1m -I0 -T -RTS --node-id core$i --topology ${topologyFile} --configuration-file $config_files/configuration.yaml --configuration-key ${configurationKey} ${ifAssetLock "--asset-lock-file ${assetLockFile}"}"
+    echo Launching core node $i: cardano-node-simple $node_args
+    cardano-node-simple $node_args &> ${stateDir}/logs/core$i.txt &
     core_pid[$i]=$!
 
   done
   for i in {1..${builtins.toString numRelayNodes}}
   do
-    node_args="--db-path ${stateDir}/relay-db''${i} --rebuild-db --listen 127.0.0.1:310''${i} --json-log ${stateDir}/logs/node''${i}.json --logs-prefix ${stateDir}/logs --system-start $system_start --metrics +RTS -N2 -qg -A1m -I0 -T -RTS --node-id relay''${i} --topology ${topologyFile} --configuration-file ${configFiles}/configuration.yaml"
-    echo Launching relay node $i with args: $node_args
-    cardano-node-simple $node_args &> /dev/null &
+    node_args="--db-path ${stateDir}/relay-db$i --rebuild-db --listen 127.0.0.1:$((3100 + i)) --json-log ${stateDir}/logs/relay$i.json --logs-prefix ${stateDir}/logs --system-start $system_start --metrics +RTS -N2 -qg -A1m -I0 -T -RTS --node-id relay$i --topology ${topologyFile} --configuration-file $config_files/configuration.yaml --configuration-key ${configurationKey}"
+    echo Launching relay node $i: cardano-node-simple $node_args
+    cardano-node-simple $node_args &> ${stateDir}/logs/relay$i.txt &
     relay_pid[$i]=$!
 
   done
   ${ifWallet ''
-    export LC_ALL=C.UTF-8
+    export LC_ALL=C
     echo Launching wallet node: ${demoWallet}
-    ${demoWallet} --runtime-args "--system-start $system_start" &> /dev/null &
+    ${demoWallet} --runtime-args "--system-start $system_start" &> ${stateDir}/logs/wallet.txt &
```

Comment:
why txt instead of log extension?

Reply:
There's a few other places where you have .txt extension to, that applies to all of them.